### ISSUE TITLE
Crashes Atom version >1.12.0

### ISSUE THREAD

**Issue by dpren** [chfritz/atom-sane-indentation#9]
Tested as only package running. From the console:

`Failed to clear temp storage: It was determined that certain files are unsafe for access within a Web application, or that too many calls are being made on file resources. SecurityError`

**Comment by chfritz** [comment]
I cannot reproduce that. For me it's working just fine with 1.12.3 (I assume you mean 1.12.0 when you say 12.0).

**Comment by dpren** [comment]
Correct, my apologies. The update resolved whatever the problem was.

**Comment by chfritz** [comment]
:+1:

**Comment by chfritz** [comment]
Now I also get it, on a different machine. Same version of atom (1.12.3). Reopening.

**Comment by dpren** [comment]
Yeah, it recurred for me too after rebuilding the package.

**Comment by chfritz** [comment]
It's really annoying. I've downgraded to 1.11 for now and will try to look into it tonight.

**Comment by chfritz** [comment]
OK, please try again now (apm update; new version = 0.1.2). I've upgraded the native package used in this package to match the version used by atom.io itself. It now seems to work again -- at least for me.

**Comment by dpren** [comment]
It worked for a little bit, but copy/pasting a block of code in the editor caused a stack overflow:

```
/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:871
Hide Stack Trace
RangeError: Maximum call stack size exceeded
    at TokenIterator.module.exports.TokenIterator.reset (/Applications/Atom.app/Contents/Resources/app.asar/src/token-iterator.js:18:9)
    at TokenizedLine.module.exports.TokenizedLine.getTokenIterator (/Applications/Atom.app/Contents/Resources/app.asar/src/tokenized-line.js:22:47)
    at TokenizedBuffer.module.exports.TokenizedBuffer.scopeDescriptorForPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/tokenized-buffer.js:540:48)
    at TextEditor.module.exports.TextEditor.scopeDescriptorForBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2837:35)
    at getRegex (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:32:16)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:56:27)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)

    ...

    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.<anonymous> (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.suggestedIndentForTokenizedLineAtBufferRow (/Users/~/.atom/packages/sane-indentation/lib/sane-indentation.js:59:30)
    at LanguageMode.module.exports.LanguageMode.suggestedIndentForBufferRow (/Applications/Atom.app/Contents/Resources/app.asar/src/language-mode.js:290:19)
    at LanguageMode.module.exports.LanguageMode.autoIndentBufferRow (/Applications/Atom.app/Contents/Resources/app.asar/src/language-mode.js:372:26)
    at TextEditor.module.exports.TextEditor.autoIndentBufferRow (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:3554:32)
    at Selection.module.exports.Selection.insertText (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:494:21)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1068:29
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1107:28
    at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:866:18)
    at TextEditor.module.exports.TextEditor.transact (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1524:26)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1101:24
    at TextEditor.module.exports.TextEditor.mergeSelections (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2498:43)
    at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2464:35)
    at TextEditor.module.exports.TextEditor.mutateSelectedText (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1099:19)
    at TextEditor.module.exports.TextEditor.insertText (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1065:19)
    at TextEditor.object.(anonymous function) [as insertText] (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/underscore-plus/lib/underscore-plus.js:77:27)
    at TextEditor.module.exports.TextEditor.insertNewline (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1080:19)
    at TextEditor.object.(anonymous function) [as insertNewline] (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/underscore-plus/lib/underscore-plus.js:77:27)
    at TextEditor.editor:newline (/Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:603:21)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:663:34
    at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:866:18)
    at TextEditor.module.exports.TextEditor.transact (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1524:26)
    at atom-text-editor.newCommandListeners.(anonymous function) (/Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:662:22)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:259:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:59
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:587:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:382:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:106:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:59)
```

I'll see if I can provide more info when I have time.

**Comment by chfritz** [comment]
ok, thanks for the stack trace.

**Comment by chfritz** [comment]
I haven't been able to reproduce this yet. Do you remember which language-mode you were in? I've tried both with a support and an unsupported language but they both worked fine for me.

**Comment by dpren** [comment]
It was just language-javascript. I haven't had that error since, so I don't know what the problem was.

**Comment by aneilbaboo** [comment]
I'm seeing the same problem using macOS Sierra 10.12.1, Atom 1.12.6 and sane-indentation 0.1.2.  I get a flash message "Auto indentation is alive! It's ALIVE!" and the Atom editor is grayed out and stops responding.

The console output isn't terribly informative: 
```
SaneIndentation was toggled!  sane-indentation.js:198
```

**Comment by chfritz** [comment]
This seems to be fixed in sane-indentation 0.2.
